frontend/src/registration/reducers.ts

```diff
@@ -1,19 +1,63 @@
 import * as RegistrationActions from './actions';
-import { blankRegistration, RegistrationModel } from './models';
+import { blankRegistrationState, RegistrationStateModel } from './models';
 
-export const initialState: Readonly<RegistrationModel> = blankRegistration;
+export const initialState: Readonly<RegistrationStateModel> = blankRegistrationState;
 
 export function registrationReducer(
   state = initialState,
   action: RegistrationActions.ActionTypes
-): RegistrationModel {
+): RegistrationStateModel {
   switch (action.type) {
+    case RegistrationActions.Types.ROOT_STATE_UPDATE:
+      // Merge state without erasing previous values
+      return {
+        loading: state.loading,
+        registrationData: {
+          ...state.registrationData,
+          ...action.data.registrationData
+        },
+        registrationFeedback: {
+          ...state.registrationFeedback,
+          ...action.data.registrationFeedback
+        }
+      };
```

Comment:
This might be a good time to include [immutability helper](https://github.com/kolodny/immutability-helper) which will reduce this to:
```javascript
return update(state, {
    registrationData: {[field]: {$set: value}},
    registrationFeedback: {[field]: {$set: '}},
});
```